client/state/notification-settings/actions.js

```diff
@@ -97,28 +95,24 @@ export const showSaveErrorNotice = () =>
 export const saveSettings = ( source, settings, applyToAll = false ) => ( dispatch ) => {
 	dispatch( { type: NOTIFICATION_SETTINGS_SAVE } );
 
-	wpcom
-		.undocumented()
-		.me()
-		.updateNotificationSettings(
-			buildSavePayload( source, settings ),
-			applyToAll,
-			( error, data ) => {
-				if ( error ) {
-					dispatch( showSaveErrorNotice() );
-					dispatch( {
-						type: NOTIFICATION_SETTINGS_SAVE_FAILED,
-						error,
-						data,
-					} );
-				} else {
-					dispatch( showSaveSuccessNotice() );
-					dispatch( {
-						type: NOTIFICATION_SETTINGS_SAVE_COMPLETE,
-						error,
-						data,
-					} );
-				}
-			}
-		);
+	const query = applyToAll ? { applyToAll: true } : {};
```

Comment:
Could we always send `applyToAll`, sometimes with a `false` value? The server PHP code checks:
```
if ( $_GET['applyToAll'] )
```
Is that falsy even if the query string value is, e.g., a `"false"` string?

Reply:
Yeah, it'll pass "false" as a string and end up being truthy. Also, the vague check on the server-side isn't very comforting, so let's keep the original behavior.

Reply:
Given the vague approach on the server currently, I’ll leave it up to you. However, if you do want to interpret `”false”` as `false`, you can use `filter_var('false', FILTER_VALIDATE_BOOLEAN)`. (See https://www.php.net/manual/en/filter.filters.validate.php)